client/src/components/DateChart.js

```diff
@@ -0,0 +1,43 @@
+import PropTypes from "prop-types"
+import React from "react"
+
+const DateChart = ({
+  items,
+  layout,
+  widgetElement: Widget,
+  viewState: viewDate,
+  widgetConfig
+}) => {
+  return (
+    <>
+      {items.map(item => {
```

Comment:
the mapping of items to widgets should is not specific to dates, It should probably happen in Chart, as it is the main function of a Chart.
This will allow us not to have a specific DateChart component. This will allow us to have an instance of Chart that then can be reconfigured from date to geo, to etc. Doing so, will allow us to reuse the same instances of the widgets and be able to animate the transitions.